tests_tf/test_utils_tf.py

```diff
@@ -0,0 +1,45 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import numpy as np
+import unittest
+import tensorflow as tf
+
+from cleverhans import utils_tf
+
+
+def numpy_kl_with_logits(q_logits, p_logits):
+    def numpy_softmax(logits):
+        exp_logits = np.exp(logits)
+        return exp_logits / np.sum(exp_logits, axis=1, keepdims=True)
+
+    q = numpy_softmax(q_logits)
+    p = numpy_softmax(p_logits)
+    return (q * (np.log(q) - np.log(p))).sum(axis=1).mean()
```

Comment:
This is not the numerically stable implementation of cross-entropy.
You need to expand
log softmax(x)_i = x_i - log sum exp(x_i)
or you can run into saturation issues.